docs/assessment.md

```diff
@@ -128,6 +128,82 @@ By default, PrairieLearn provides an incentive for students to answer a specific
+6 | 0% | 50% | 4 | 8 | No improvement in score, so no points awarded
+7 | 90% | 90% | 4 | 9.6 | (90% - 50%) * 4 = 1.6 points awarded
+8 | 100% | 0% | 8 | 10 | (100% - 90%) * 4 = 0.4 points awarded, `best current score` reset to 0%, `question value` increased to 8
+
+After the eight submissions above the student has achieved maximum points for the question. They can continue to answer the question for additional practice but they cannot earn any more points for it.
+
+## Question scoring for `Exam` assessments
+
+Assessments with `"type": "Exam"` are designed to test students' knowledge of course material. Students can retry questions for reduced points, so long as `allowRealTimeGrading` is true (the default). This section describes in detail how repeated attempts are scored, assuming that the question has `autoPoints` set to an array of values.
+
+The important question configuration settings is:
+- **`autoPoints`** is an array of points, with the `n`-th entry being the number of points that a student receives for a fully correct submission on the `n`-th attempt at the question. If `autoPoints` is set to a single value rather than an array then only a single attempt is allowed.
+
+The basic idea is that `autoPoints` is set to a decreasing sequence of points, such as `[10, 7, 5, 2]`. In this case, a correct answer on the first attempt is worth 10 points, a correct answer on the second attempt is worth 7 points, and so on. If the student does not answer the question correctly after four attempts, they will receive 0 points for the question and cannot attempt it again.
+
+The details of how scoring works are made more complex by the fact that a student may receive partial credit for any given submission attempt. PrairieLearn uses a scoring system that has four key properties, assuming that `autoPoints` is an array of decreasing values:
+1. A student can never lose points by attempting a question again.
+2. If a student submits a more correct answer than their previous answers, they will always receive some additional points for it.
+3. Taking more attempts to answer correctly will always be worth less points than answering in fewer attempts, so students are always incentivized to answer questions correctly as soon as possible.
+4. The total number of submission attempts that a student can make is limited by the number of entries in `autoPoints`.
+
+The exact details of how PrairieLearn awards points are as follows. First, a number of variables are computed internally by PrairieLearn as the student answers the question:
+- **Submission percentage score** is the score (between 0% and 100%) that the student receives for any given submission attempt, as evaluated by the question grading specification.
+- **Best current score** is the highest submission percentage score that the student has received since last achieving 100% for a submission. This is initialized to zero.
```

Comment:
```suggestion
- **Best current score** is the highest submission percentage score that the student so far. This is initialized to zero.
```
This is exam, once you get 100% you are done.

Reply:
Done in 4191a4335164ac391314b4b9f76d725eba219f17